File: 백준/Silver/1676. 팩토리얼 0의 개수/팩토리얼 0의 개수.py
n=int(input())
s=0
if n==0:
    print(0)
else:
    for i in range(n-1,1,-1):
        n=n*i
    cnt=list(str(n))
    cnt=cnt[-1::-1]
    for c in cnt:
        if c!='0':
            break
        s+=1
    print(s)

File: 백준/Silver/1676. 팩토리얼 0의 개수/팩토리얼 0의 개수.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.math.BigInteger;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        BigInteger big = new BigInteger("1");
        int cnt = 0;
        for (int i = 1; i <= N; i++) {
            big = big.multiply(BigInteger.valueOf(i));
        }

        String str = big + "";
        for (int i = str.length() - 1; i >= 0; i--) {
            if (str.charAt(i) == '0') cnt++;
            else break;
        }
        System.out.println(cnt);
    }
}